### PR TITLE
World114 update

### PR DESCRIPTION
**Pull request by miclav**
This pull request contains spritemap updated for the 1.14 update, and includes the quick patch from Moritz Neikes  (https://github.com/mapcrafter/mapcrafter/issues/317#issuecomment-468392202)  for chunk loading 1.14.
Some blocks needed special code for the update to work.
- Composter
- Bamboo
- Signs
The only blocks not fully implemented yet is the bell, it's only displaying the bell support.
image

### CODE REVIEW

File: src/mapcraftercore/mc/chunk.cpp [hunk 1]
```diff
@@ -173,7 +173,7 @@ bool Chunk::readNBT(mc::BlockStateRegistry& block_registry, const char* data, si
 		const nbt::TagString& tag = level.findTag<nbt::TagString>("Status");
 		// completely generated chunks in fresh 1.13 worlds usually have status 'fullchunk' or 'postprocessed'
 		// however, chunks of converted <1.13 worlds don't use these, but the state 'mobs_spawned'
-		if (!(tag.payload == "fullchunk" || tag.payload == "postprocessed" || tag.payload == "mobs_spawned")) {
+               if (!(tag.payload == "fullchunk" || tag.payload == "full" || tag.payload == "postprocessed" || tag.payload == "mobs_spawned")) {
```
Comment (m0r13) [inline src/mapcraftercore/mc/chunk.cpp]: Please indent this with tabs.

File: src/mapcraftercore/mc/chunk.cpp [hunk 2]
```diff
@@ -273,8 +268,16 @@ bool Chunk::readNBT(mc::BlockStateRegistry& block_registry, const char* data, si
 			continue;
 		}
 
-		std::copy(block_light.payload.begin(), block_light.payload.end(), section.block_light);
-		std::copy(sky_light.payload.begin(), sky_light.payload.end(), section.sky_light);
+               if(section_tag.hasArray<nbt::TagByteArray>("BlockLight", 2048))
```
Comment (m0r13) [inline src/mapcraftercore/mc/chunk.cpp]: Please use tabs for indentation here as well. And look how I placed the curly brackets in the remaining code, also note the space between `if` and the condition.

File: src/mapcraftercore/renderer/blocktextures.cpp [hunk 3]
```diff
@@ -11,33 +11,112 @@ namespace mapcrafter {
 namespace renderer {
 
 BlockTextures::BlockTextures()
-	: ANVIL_BASE("anvil_base"),
-	  ANVIL_TOP_DAMAGED_0("anvil_top_damaged_0"),
-	  ANVIL_TOP_DAMAGED_1("anvil_top_damaged_1"),
-	  ANVIL_TOP_DAMAGED_2("anvil_top_damaged_2"),
+	: ACACIA_DOOR_BOTTOM("acacia_door_bottom"),
```
Comment (m0r13) [inline src/mapcraftercore/renderer/blocktextures.cpp:14]: Just FYI, this doesn't need to be updated as Mapcrafter doesn't need the Minecraft texture files anymore (but only the generated block images instead). We can keep it as it is now though.

**Review by m0r13 — CHANGES_REQUESTED** [review]
Just two little things.

**Comment by m0r13** [comment]
Hey. Thanks for helping out and making the PR! I looked at the code... Just two little things then I would merge it and see that I build some binaries for that version again.

**Comment by miclav** [comment]
Hi
Sure thing, I'll work on those tonight. Sorry about the tab indent, I should have known better :(
I will send as well a PR for blockcrafter, the changes to add missing blockstates.

**Comment by miclav** [comment]
Changes made

**Comment by lukruh** [comment]
Hi @miclav, thanks for adding 1.14 support so fast. Did something changed in the build process? Some new dependencies? I tryed to build your world114 branch as described [in the docs for debian](https://docs.mapcrafter.org/builds/stable/installation.html#general-instructions)  ended up with errors in `make`:

    Scanning dependencies of target mapcrafter                                                                                                                                                                          
    [ 75%] Building CXX object CMakeFiles/mapcrafter.dir/mapcrafter.o                                                                                                                                                   
    [ 76%] Linking CXX executable mapcrafter                                                                                                                                                                            
    CMakeFiles/mapcrafter.dir/mapcrafter.o: In function `main':                                                                                                                                                         
    mapcrafter.cpp:(.text+0x143): undefined reference to `boost::program_options::options_description::m_default_line_length'
    .... many more "undefined references"

probably a trivial problem :S I'm not very used to c++ and compiling stuff, but it worked a while ago for the world113 branch, thats why I'm asking here... 
thanks a lot

**Comment by beulette** [comment]
I had to add set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11") in CMakeList.txt

Or I had an issue when launching "make":
"#error This file requires compiler and library support for the ISO C++ 2011 standard".

And then I compiled til 62% with this error (at Building CXX object mapcraftercore/CMakeFiles/mapcraftercore.dir/util/logging.o):
```
###/src/mapcraftercore/util/logging.cpp: In member function ‘virtual void mapcrafter::util::LogSyslogSink::sink(const mapcrafter::util::LogMessage&)’:
###/src/mapcraftercore/util/logging.cpp:269:56: warning: format not a string literal and no format arguments [-Wformat-security]
    util::replaceAll(message.message, "%", "%%").c_str());
```

And then at 71% ([ 71%] Linking CXX executable mapcrafter) I had a lot of "undefined references" just like sarbot.

I just found here that someone says "There were changes to the <string> class in the C++11 standard which may conflict with versions of the Boost library that were compiled with a non-C++11 compiler (such as G++-4.8). Try recompiling boost or using a version of C++ compiler that was used to compile your Boost libraries." (source: https://stackoverflow.com/questions/12179154/undefined-reference-to-boostprogram-optionsoptions-descriptionm-default-l)

**Comment by miclav** [comment]
Hey @sarbot 
To answer your question, I didn't change anything in the project but few lines of code and resource files. The build process for world114 should be identical to world113.
And thanks @beulette for troubleshooting and clarifying the compiling issue.

**Comment by orion78fr** [comment]
Using `cmake . && make` on this branch worked perfectly for my 1.14 map on Arch Linux. I had residual files from the 1.13 branch, so it may be that.

**Comment by 25A0** [comment]
Hi @miclav, I just want to thank you for turning my patch into a proper PR, and for the other work you've done here! 🎉

**Review by m0r13 — APPROVED** [review]

**Comment by m0r13** [comment]
Looks good now! I'll merge it and see that I'll trigger some builds too.

**Comment by lukruh** [comment]
[100%] Built target mapcrafter

It worked, while trying to update my libboost package I realized that a lot of packages have been outdated reinstalling the dependencies after `apt update` solved. Thanks @beulette , my friends will be happy to enjoy this awesome map again :)

**Pull request merged**